### PR TITLE
the persisting can now be used with session variables instead of cookies

### PR DESCRIPTION
**Pull request by marineusde**
…okies

## ⚡ PowerGrid - Pull Request

Welcome and thank you for your interest in contributing to our project!. You must use this template to submit a Pull Request or it will not be accepted.

--- 
#### Motivation

- [ ] Bug fix
- [ ] Enhancement
- [X] New feature
- [ ] Breaking change

#### Description

the developer can configure, that the persisting works with session storage instead of cookies

#### Documentation

 This PR requires [Documentation](https://github.com/Power-Components/powergrid-doc) update?

- [ ] Yes
- [X] No
- [ ] I have already submitted a Documentation pull request.

### CODE REVIEW

File: resources/config/livewire-powergrid.php [hunk 1]
```diff
@@ -79,6 +79,19 @@
...
+    | Persisting
+    |--------------------------------------------------------------------------
+    |
+    | PowerGrid supports persisting of the filters, columns and sorting.
+    | 'true': persist in the session.
+    | 'false': persist with cookies (default).
+    |
+    */
+
+    'persist_in_session' => false,
```
Comment (luanfreitasdev) [inline resources/config/livewire-powergrid.php]: What do you think of `'persist_driver' => 'cookies' // session`? It is easier to implement if there is another

**Review by luanfreitasdev — CHANGES_REQUESTED** [review]
Thank you for this PR!

Can you add some tests?

File: resources/config/livewire-powergrid.php (hunk 1 above)
Comment (marineusde) [reply resources/config/livewire-powergrid.php]: good idea, no problem

Comment (marineusde) [reply resources/config/livewire-powergrid.php]: Its changed. I will write the test tomorrow for the full trait. Whats the name of the right folder in test/ for the test-file?

Comment (luanfreitasdev) [reply resources/config/livewire-powergrid.php]: I've added todo tests in https://github.com/Power-Components/livewire-powergrid/pull/1509/commits/76aaa26ada39cab8cd7c349cfdbe7718baef7c21

Comment (marineusde) [reply resources/config/livewire-powergrid.php]: done

can you tell me please, why I cant see the code coverage in the trait? I tried it with: "./vendor/pestphp/pest/bin/pest --coverage tests/Feature/PersistTest.php"

**Comment by marineusde** [comment]
Do you have any other suggestions for the pull request? :)

**Review by luanfreitasdev — APPROVED** [review]

**Comment by luanfreitasdev** [comment]
Thank you @marineusde

**Pull request merged**